lib/ansible/modules/cloud/amazon/ec2_instance.py

```diff
@@ -1130,7 +1139,11 @@ def build_top_level_options(params):
     if params.get('placement_group'):
         spec.setdefault('Placement', {'GroupName': str(params.get('placement_group'))})
     if params.get('ebs_optimized') is not None:
-        spec['EbsOptimized'] = params.get('ebs_optimized')
+        if params.get('ebs_optimized') == 'auto':
+            ebs_optimized = True if _supports_ebs_optimized(params.get('instance_type')) else False
```

Comment:
```suggestion
            ebs_optimized = _supports_ebs_optimized(params.get('instance_type'))
```

Reply:
erm.. Not sure why I put that there. Fixed, thanks! :)